integration/bundle/init_default_python_test.go

```diff
@@ -0,0 +1,114 @@
+package bundle_test
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/databricks/cli/integration/internal/acc"
+	"github.com/databricks/cli/internal/testcli"
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/databricks/cli/libs/python/pythontest"
+	"github.com/stretchr/testify/require"
+)
+
+var pythonVersions = []string{
+	"3.8",
+	"3.9",
+	"3.10",
+	"3.11",
+	"3.12",
+	"3.13",
+}
+
+var pythonVersionsShort = []string{
+	"3.9",
+	"3.12",
+}
+
+var extraInstalls = map[string][]string{
+	"3.12": {"setuptools"},
+	"3.13": {"setuptools"},
+}
+
+func TestDefaultPython(t *testing.T) {
+	versions := pythonVersions
+	if testing.Short() {
+		versions = pythonVersionsShort
+	}
+
+	for _, pythonVersion := range versions {
+		t.Run(pythonVersion, func(t *testing.T) {
+			testDefaultPython(t, pythonVersion)
+		})
+	}
+}
+
+func testDefaultPython(t *testing.T, pythonVersion string) {
+	ctx, wt := acc.WorkspaceTest(t)
+
+	uniqueProjectId := testutil.RandomName("")
+	ctx, replacements := testcli.WithReplacementsMap(ctx)
+	replacements.Set(uniqueProjectId, "$UNIQUE_PRJ")
+
+	testcli.PrepareReplacements(t, replacements, wt.W)
+
+	user, err := wt.W.CurrentUser.Me(ctx)
+	require.NoError(t, err)
+	require.NotNil(t, user)
+	testcli.PrepareReplacementsUser(t, replacements, *user)
+
+	tmpDir := t.TempDir()
+	testutil.Chdir(t, tmpDir)
+
+	opts := pythontest.VenvOpts{
+		PythonVersion: pythonVersion,
+		Dir:           tmpDir,
+	}
+
+	pythontest.RequireActivatedPythonEnv(t, ctx, &opts)
+	extras, ok := extraInstalls[pythonVersion]
+	if ok {
+		args := append([]string{"pip", "install", "--python", opts.PythonExe}, extras...)
+		cmd := exec.Command("uv", args...)
+		require.NoError(t, cmd.Run())
+	}
+
+	projectName := "project_name_" + uniqueProjectId
+
+	initConfig := map[string]string{
+		"project_name":     projectName,
+		"include_notebook": "yes",
+		"include_python":   "yes",
+		"include_dlt":      "yes",
+	}
+	b, err := json.Marshal(initConfig)
+	require.NoError(t, err)
+	err = os.WriteFile(filepath.Join(tmpDir, "config.json"), b, 0o644)
+	require.NoError(t, err)
+
+	testcli.RequireOutput(t, ctx, []string{"bundle", "init", "default-python", "--config-file", "config.json"}, "testdata/default_python/bundle_init.txt")
+	testutil.Chdir(t, projectName)
+
+	testcli.RequireOutput(t, ctx, []string{"bundle", "validate"}, "testdata/default_python/bundle_validate.txt")
+
+	testcli.RequireOutput(t, ctx, []string{"bundle", "deploy"}, "testdata/default_python/bundle_deploy.txt")
+	t.Cleanup(func() {
```

Comment:
Shouldn't the cleanup be set before running any of the comments above?

Reply:
You mean before validate, because of mkdir call in "validate"? I guess so.

Note, the function is not named correctly. It is actually asserting on the output, not requiring it. I renamed it to AssertOutput.

I also moved t.Cleanup before validate().